packages/ui/src/ui/containers/ACL/RequestPermissions/RequestPermissions.tsx

```diff
@@ -122,9 +122,31 @@ function RequestPermissions(props: Props) {
     );
 
     const currentCaption = `Current ${SHORT_TITLE[idmKind] ?? idmKind}`;
-    const {permissionsToRequest: choices} = UIFactory.getAclPermissionsSettings()[idmKind];
+    const {permissionsToRequest, getAvailablePermissions} =
+        UIFactory.getAclPermissionsSettings()[idmKind];
+
+    const [availablePermissions, setAvailablePermissions] = React.useState<
+        typeof permissionsToRequest | undefined
+    >(undefined);
+
+    const onShow = useCallback(async () => {
+        if (!getAvailablePermissions) {
+            handleShow();
```

Comment:
here you can add return;. Then you won't need "else" and the nesting will be slightly reduced